libsolidity/ast/AST.h

```diff
@@ -522,10 +522,18 @@ class ContractDefinition: public Declaration, public StructurallyDocumented, pub
 	std::vector<EventDefinition const*> events() const { return filteredNodes<EventDefinition>(m_subNodes); }
 	std::vector<EventDefinition const*> const& definedInterfaceEvents() const;
 	std::vector<EventDefinition const*> const usedInterfaceEvents() const;
+	/// @return all events defined in this contract and its base contracts and all events
+	/// that are emitted during the execution of the contract.
+	/// @param _requireCallGraph if false, do not fail if the call graph has not been computed yet.
+	std::vector<EventDefinition const*> interfaceEvents(bool _requireCallGraph = true) const;
 	/// @returns all errors defined in this contract or any base contract
 	/// and all errors referenced during execution.
 	/// @param _requireCallGraph if false, do not fail if the call graph has not been computed yet.
 	std::vector<ErrorDefinition const*> interfaceErrors(bool _requireCallGraph = true) const;
+	/// @return same as interfaceEvents, but allowing only unique signatures
+	/// In case of conflict between a library event and a contract one, selects the later
+	/// In case of conflict between two library events, selects the one that is defined first
+	std::vector<EventDefinition const*> natspecInterfaceEvents() const;
```

Comment:
I don't think this name fully conveys why this function is here. Maybe something like this would be better:
```suggestion
	std::vector<EventDefinition const*> uniqueInterfaceEvents() const;
```

But overall, this naming problem reflects the fact that it's doing something very specific and non-universal and its existence is mostly a hack. Selecting one of the events in case of a conflict is not something we want to be doing long-term. We do it only for Natspec and then only until we can change the fields to lists in v2. For that reason, I think we'd be better off keeping this code bottled in `Natspec.cpp`, in the one function that actually needs it. No need to make it more prominent and reusable.